File: Back-End-M/src/main/java/com/attendance/fpt/services/OtpService.java
package com.attendance.fpt.services;

import com.attendance.fpt.model.request.ResetPasswordWithOtpRequest;
import com.attendance.fpt.model.response.OtpResponse;

public interface OtpService {
    void sendOtp(String email);

    OtpResponse verifyOtp(String email, String otpCode);

    void resetPasswordWithOtp(ResetPasswordWithOtpRequest resetPasswordWithOtpRequest);
}

File: Front-End/src/services/otp.service.ts
import http from "@/utils/http";

export const otpApi = {
  sendOtp: async (email: string) => {
    const response = await http.post("/otp/send", { email });
    return response.data;
  },

  verifyOtp: async (email: string, otpCode: string) => {
    const response = await http.post("/otp/verify", { email, otpCode });
    return response.data;
  },

  resetPasswordWithOtp: async (request: any) => {
    const response = await http.post("/otp/reset-password", request);
    return response.data;
  },
};
